### PR TITLE
feature/64166 Add work package internal comments (write) API

### PR DESCRIPTION
**Pull request by akabiru**
# Ticket

https://community.openproject.org/work_packages/64166

# What are you trying to accomplish?

Support "internal" flag in `POST /api/v3/work_packages/:id/activities` API that is responsible for creating wp comments

## Screenshots

Screenshot 2025-05-23 at 11 13 07 AM
Screenshot 2025-05-23 at 11 12 57 AM

# Merge checklist

- [x] Added/updated tests
- [x] Added/updated documentation in Lookbook (patterns, previews, etc)
- [ ] Tested major browsers (Chrome, Firefox, Edge, ...)

### CODE REVIEW

File: docs/api/apiv3/components/schemas/activity_comment_write_model.yml [hunk 1]
```diff
@@ -3,4 +3,14 @@
...
   comment:
-    type: string
+    type: object
+    properties:
+      raw:
+        type: string
+        description: The raw content of the comment
+  internal:
+    type: boolean
+    description: Whether this comment is internal (only available to users with add_internal_comments permission)
```
Comment (Kharonus) [inline docs/api/apiv3/components/schemas/activity_comment_write_model.yml]: 🟡 Can you add maybe the default value here? Explaining, that if the client doesn't send the properties it goes to `false`?

File: spec/requests/api/v3/activities_by_work_package_resource_spec.rb [hunk 2]
```diff
@@ -148,6 +148,56 @@
             .at_path("message")
         end
       end
+
+      context "when creating an internal comment" do
+        shared_context "to create internal comment" do |internal: false|
```
Comment (Kharonus) [inline spec/requests/api/v3/activities_by_work_package_resource_spec.rb:153]: 🟡 related to the previous comment, can you add a single test case where `internal` is `nil` and thus going to default?

File: docs/api/apiv3/components/schemas/activity_comment_write_model.yml (hunk 1 above)
Comment (Kharonus) [reply docs/api/apiv3/components/schemas/activity_comment_write_model.yml]: so maybe:

```yaml
  internal:
    type: boolean
    description: |-
      Determines whether this comment is internal. This is only available to users
      with `add_internal_comments permission`. It defaults to false, if unset.
    default: false
```

**Review by Kharonus — CHANGES_REQUESTED** [review]
some minor improvement requests :D

**Review by Kharonus — APPROVED** [review]
nice, LGTM!
